ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
 
-The `Player Prefab` field inside of Boss Room's `NetworkManager` is populated with `PersistentPlayer` prefab. Netcode will spawn a PersistentPlayer per client connection, with the client designated as the owner of the prefab instance. All `Player Prefab` prefab instances will be migrated between scenes internally by Netcode's scene management, therefore it is not necessary to mark this object as a `DontDestroyOnLoad` object. This object is suitable for storing data, in some cases in the form of `NetworkVariable`s, that could be accessed across scenes (eg. name, avatar GUID, etc). PersistentPlayer's GameObject hierarchy is quite trivial as it is comprised of only one GameObject:
+The purpose of `Persistent Player` is to store synchronized data about player, such as their name, chosen avatar GUID etc.
 
-* PersistentPlayer: a `NetworkObject` that will not be destroyed between scenes
+This `PlayerAvatar` prefab instance, that is owned by the corresponding connected client, is destroyed by Netcode when a scene load occurs (either to `PostGame` scene, or back to `MainMenu` scene), or through client disconnection.
 
-####CharSelect Scene
 Inside `CharSelect` scene, clients select from 8 possible avatar classes, and that selection is stored inside PersistentPlayer's `NetworkAvatarGuidState`.
 
-####BossRoom Scene
-Inside `BossRoom` scene, `ServerBossRoomState` spawns a `PlayerAvatar` per PersistentPlayer present. This `PlayerAvatar` prefab instance, that is owned by the corresponding connected client, is destroyed by Netcode when a scene load occurs (either to `PostGame` scene, or back to `MainMenu` scene), or through client disconnection.
+Inside `BossRoom` scene, `ServerBossRoomState` spawns a `PlayerAvatar` per PersistentPlayer present.
+
+`ClientAvatarGuidHandler`, a `NetworkBehaviour` component residing on the `PlayerAvatar` prefab instance will fetch the validated avatar GUID from `NetworkAvatarGuidState`, and spawn a local, non-networked graphics GameObject corresponding to the avatar GUID.
+
+### Characters
+
+`ServerCharacter` contains NetworkVariables that store the state of any given character, and the server RPCs. It is responsible for executing or kicking off the server-side logic for the characters, which includes:
+
+- movement and pathfinding via `ServerCharacterMovement` - it uses NavMeshAgent that lives on the server to translate the character’s transform, which is synchronized using NetworkTransform component;
+- player action queueing and execution via `ServerActionPlayer`;
+- AI logic via `AIBrain` (applies to NPCs);
+- Character animations via `ServerAnimationHandler`, which themselves are synchronized using NetworkAnimator;
+
+`ClientCharacter` primarily is a host for the `ClientActionPlayer` class, and it also contains the client RPCs for the character gameplay logic.
+
+### Game data setup
+
+Game data in Boss Room is defined in `ScriptableObjects`.
```

Comment:
```suggestion
### Game config setup

Game config in Boss Room is defined in `ScriptableObjects`.
```
Didn't we end up using "config" instead of data? I still see "data" in our source code though :/